File: 2022/D6/tuning.py
import time

start = time.time()
f = "input.txt"
line = open(f).read().strip()

def streamcheck(length: int) -> int:
    for i in range(len(line)):
        mark = line[i:i+length]
        if len(set(mark)) == length:
            return i+length
        
print(f"Part 1: {streamcheck(4)}")
print(f"Part 2: {streamcheck(14)}")

print("Total Time: " + str(time.time() - start))

File: 2022/D6/tuning.go
	package main

import (
	"fmt"
	"strings"
	"time"

	"github.com/Galzzly/AoC/utils"
)

func main() {
	start := time.Now()
	f := "input.txt"

	stream := utils.ReadFileSingleLine(f)
	t1 := time.Now()
	fmt.Println("Part 1:", streamcheck(stream, 4), "- Took", time.Since(t1))
	t2 := time.Now()
	fmt.Println("Part 2:", streamcheck(stream, 14), "Took:", time.Since(t2))
	fmt.Println(time.Since(start))
}

func streamcheck(stream string, length int) (res int) {
	for i := 0; i < len(stream)-length; i++ {
		if checkUnique(stream[i : i+length]) {
			return i + length
		}
	}
	return
}

func checkUnique(str string) bool {
	for _, c := range str {
		if strings.Count(str, string(c)) > 1 {
			return false
		}
	}
	return true
}
